scripts/setup-repo.sh

```
@@ -0,0 +1,207 @@
+      # template: any rule the live one has and the template lacks would be silently
+      # switched off. sreforge carries required_review_thread_resolution and
+      # require_extra_approval_for_unattributed_changes, and on a contract with zero
+      # required approvals thread resolution is the only thing enforcing a finding.
+      say "  \"$RULESET_NAME\" exists (id $EXISTING). Not touching it."
+      LIVE=$(printf '%s' "$RULESETS" | jq --arg n "$RULESET_NAME" '.[]|select(.name==$n)')
+      LIVE_FULL=$(gh api "repos/$REPO/rulesets/$EXISTING" 2>/dev/null || printf '%s' "$LIVE")
+      LIVE_TYPES=$(printf '%s' "$LIVE_FULL" | jq -r '[.rules[]?.type]|sort|join(",")')
+      WANT_TYPES=$(printf '%s' "$BODY" | jq -r '[.rules[]?.type]|sort|join(",")')
+      say "  live rules:  $LIVE_TYPES"
+      say "  template:    $WANT_TYPES"
+      [ "$LIVE_TYPES" = "$WANT_TYPES" ] && say "  rule types match" || say "  RULE TYPES DIFFER, review by hand"
```

Shellcheck (0.11.0)

[info] Note that A && B || C is not if-then-else. C may run when A is true.

(SC2015)